### PR TITLE
test: Add comprehensive test coverage for FormTextEntity component

### PR DESCRIPTION
**Pull request by aasimsyed**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
- [x] I have run the tests for the backend and frontend depending on what's needed for my changes as described in the [testing section of the contributing guide](CONTRIBUTING.md#testing)

---

## Summary

Adds comprehensive unit test coverage for the `FormTextEntity` component and fixes validation bugs discovered during testing.

### Bug Fixes
- **Fixed max length error message**: Corrected error message to show "2500" instead of "25000" characters
- **Fixed custom error message override**: Ensured custom "description required" message displays instead of generic "Required" message by using `.refine()` with `.default("")` in Zod schema
- **Improved whitespace URL validation**: Whitespace-only URLs are now treated as empty, triggering the "text or URL required" error instead of invalid URL error

### Test Coverage
- **59 comprehensive unit tests** covering:
  - Basic rendering (with/without props)
  - Text validation (description, getInvolved, URL validation)
  - Change events and form data initialization
  - Dynamic props and form submission
  - Style coverage (error states, class changes)
  - Accessibility (ARIA attributes, labels, focus handling)
  - Edge cases (long inputs, special characters, Unicode, URL variations)

### New Edge Case Tests
- Description with only whitespace validation
- URL without protocol rejection
- Complete `handleSubmit` payload verification
- Optional fields handling (undefined vs empty string)

## Testing

All 59 tests pass, covering:
- Logic validation and error handling
- Accessibility requirements
- Edge cases and boundary conditions
- Form submission behavior

## Files Changed

- `frontend/app/components/form/FormTextEntity.vue` - Validation schema improvements
- `frontend/test/components/form/FormTextEntity.spec.ts` - Comprehensive test suite

### Related issue

- closes https://github.com/activist-org/activist/issues/1759

### CODE REVIEW

File: frontend/app/components/form/FormTextEntity.vue [hunk 1]
```diff
@@ -80,13 +80,15 @@ const schema = z
...
       .string()
-      .min(1, t("i18n.components.form_text_entity.description_required"))
-      .max(
-        2500,
-        t("i18n.components.form_text_entity.max_text_length", {
-          max_text_length: 25000,
-        })
-      ),
+      .default("")
+      .refine((val) => val.trim().length > 0, {
```
Comment (nicki182) [inline frontend/app/components/form/FormTextEntity.vue]: note: I think you can trim first and then do min and max

Comment (aasimsyed) [reply frontend/app/components/form/FormTextEntity.vue]: praise: Thank you for the review, @nicki182!

### Changes Made

I've implemented your suggestion to use Zod's built-in `.min()` and `.max()` methods with trimming:

**Before:**
```typescript
description: z
  .string()
  .default("")
  .refine((val) => val.trim().length > 0, { ... })
  .refine((val) => val.length <= 2500, { ... })
```

**After:**
```typescript
description: z
  .string()
  .default("")
  .transform((val) => val.trim())
  .pipe(
    z
      .string()
      .min(1, t("i18n.components.form_text_entity.description_required"))
      .max(
        2500,
        t("i18n.components.form_text_entity.max_text_length", {
          max_text_length: 2500,
        })
      )
  )
```

**Review by andrewtavis — APPROVED** [review]
praise: All's looking good here, @aasimsyed :) Thanks for how comprehensive all of this is!

note: Checked and specifically we are checking for cases where the `getInvolved` text and the `getInvolvedUrl` are missing, which is the main edge case.
